lib/templates/completion.zsh

```diff
@@ -1,3 +1,4 @@
+#compdef {pkgname}
 ###-begin-{pkgname}-completion-###
```

Comment:
The `###-begin-{pkgname}-completion-###` and `###-end-{pkgname}-completion-###` are there to determine the range of generate code should it be inserted to `.zshrc`. I have some questions:
1. Must `compdef` strictly be at the top?
2. Would it affect the `pnpm completion zsh >> ~/.zshrc` use-case?
3. Would it affect the `pnpm completion zsh > pnpm-completion.zsh` + `source pnpm-completion.zsh` use case?

Reply:
1. `compdef` must be the first line in the file, for it to be recognized by the completion autoload system, see https://zsh.sourceforge.io/Doc/Release/Completion-System.html#Autoloaded-files for reference.
2. Depends on what the condition on `zsh_eval_context` evaluates to when zsh sources `.zshrc`, if it doesn't work, we can try a different condition, such as `[[ $funcstack[1] = _{pkgname}_completion ]]` that some other such completion files use.
3. That should still work.